src/ui/public/state_management/state_monitor_factory.js

```diff
@@ -0,0 +1,103 @@
+import { cloneDeep, isEqual, set, isPlainObject } from 'lodash';
+
+export default {
+  create: (state, customInitialState) => stateMonitor(state, customInitialState)
+};
+
+function stateMonitor(state, customInitialState) {
+  let destroyed = false;
+  let ignoredProps = [];
+  let changeHandlers = [];
+  let initialState;
+
+  setInitialState(customInitialState);
+
+  function setInitialState(customInitialState) {
+    // state.toJSON returns a reference, clone so we can mutate it safely
+    initialState = cloneDeep(customInitialState) || cloneDeep(state.toJSON());
+  }
+
+  function removeIgnoredProps(state) {
+    ignoredProps.forEach(path => {
+      set(state, path, true);
+    });
+    return state;
+  }
+
+  function getStatus() {
+    const currentState = removeIgnoredProps(cloneDeep(state.toJSON()));
```

Comment:
Can you add this above this line: `// state.toJSON returns a reference, clone so we can mutate it safely`